grants_tagger/bertmesh/model.py

```diff
@@ -0,0 +1,45 @@
+from transformers import AutoModel
+import torch
+
+
+class MultiLabelAttention(torch.nn.Module):
+    def __init__(self, D_in, num_labels):
+        super().__init__()
+        self.A = torch.nn.Parameter(torch.empty(D_in, num_labels))
+        torch.nn.init.uniform(self.A, -0.1, 0.1)
+
+    def forward(self, x):
+        attention_weights = torch.nn.functional.softmax(
+            torch.tanh(torch.matmul(x, self.A)), dim=1
+        )
+        return torch.matmul(torch.transpose(attention_weights, 2, 1), x)
+
+
+class BertMesh(torch.nn.Module):
+    def __init__(self, pretrained_model, num_labels, multilabel_attention=False):
+        super().__init__()
+        self.pretrained_model = pretrained_model
+        self.num_labels = num_labels
+        self.multilabel_attention = multilabel_attention
+
+        self.bert = AutoModel.from_pretrained(pretrained_model)  # 768
+        self.multilabel_attention = MultiLabelAttention(
+            768, num_labels
+        )  # num_labels, 768
+        self.linear_1 = torch.nn.Linear(768, 512)  # num_labels, 512
```

Comment:
consider using the sequential api for this?
```nn.Sequential(...)```

Reply:
👍

Reply:
actually in this case we use different layers depending on whether we add a multilabel attention or not. not all are used so this reduces duplication but generally good shout